test/framework/workload.go

```diff
@@ -63,6 +67,42 @@ func (w *WorkloadCluster) WaitForKubeconfig() {
 	if err != nil {
 		w.T.Fatalf("Failed waiting for cluster kubeconfig: %s", err)
 	}
+	w.initClusterStateConfig(ctx)
+}
+
+// ValidateClusterDelete verifies the cluster has been deleted.
+func (w *WorkloadCluster) ValidateClusterDelete() {
+	ctx := context.Background()
+	w.T.Logf("Validating cluster deletion %s", w.ClusterName)
+	if w.clusterStateConfig == nil {
```

Comment:
What is check for?

Reply:
When `clusterStateConfig` is required for this to work. Are you thinking it's better to leave out this check because it should error in that case?